PythonVisualizations/SimpleQueue.py

```diff
@@ -6,357 +6,238 @@
-        canvas.delete(n.display_shape)
-        canvas.delete(n.display_val)
-        
-        self.onOffButtons()
-        
-        # update window
-        window.update()
-    
-    def removeRear(self):
-        
-        #if the queue is empty, exit
-        if self.size == 0:
-            return None
-        
-        #get the value at rear, and then set it to None so it will be garbage collected
-        n = self.list[self.rear]
-        self.list[self.rear] = None
-        
-        #decrement rear
-        self.rear -= 1
-        
-        #deal with wraparound
-        if self.rear == -1:
-            self.rear = self.size-1
-            
+            self.moveItemsBy(self.frontArrow, (self.CELL_SIZE * self.size * -1, 0))
+        #move arrow
+        self.moveItemsBy(self.frontArrow, (self.CELL_SIZE, 0))
+        #decrement number of arrows
         self.nItems -= 1
-        
-        # delete the associated display objects
-        canvas.delete(n.display_shape)
-        canvas.delete(n.display_val)
-        
-        self.onOffButtons()
-        
+
         # update window
-        window.update()
-    
+        self.window.update()
+        self.cleanUp(callEnviron)
+
     def display(self):
-        canvas.delete("all")
-        xpos = ARRAY_X0
-        ypos = ARRAY_Y0
 
-        for i in range(self.size):
-            canvas.create_rectangle(xpos + CELL_SIZE*i, ypos, xpos + CELL_SIZE*(i+1), ypos + CELL_SIZE, fill='white', outline='black')
+        self.canvas.delete("all")
+
+        self.makeButtons()
+        self.onOffButtons()
+        xpos = self.ARRAY_X0
+        ypos = self.ARRAY_Y0
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
 
         # go through each Element in the list
         for n in self.list:
-            
+
             # Only loop through the existing elements
             if n:
-                print(n)
+
                 # create display objects for the associated Elements
-                cell = canvas.create_rectangle(xpos, ypos, xpos+CELL_SIZE, ypos+CELL_SIZE, fill=n[1], outline='')
-                cell_val = canvas.create_text(xpos+(CELL_SIZE/2), ypos+(CELL_SIZE/2), text=n[0], font=('Helvetica', '20'))
-    
+                cell = self.canvas.create_rectangle(xpos, ypos, xpos+self.CELL_SIZE, ypos+self.CELL_SIZE, fill=n[1], outline='')
+                cell_val = self.canvas.create_text(xpos+(self.CELL_SIZE/2), ypos+(self.CELL_SIZE/2), text=n[0], font=('Helvetica', '20'))
+
                 # save the display objects to the appropriate attributes of the Element object
                 n.display_shape = cell
                 n.display_val = cell_val
-    
+
                 # increment xpos
-                xpos += CELL_SIZE
+                xpos += self.CELL_SIZE
 
-        window.update()
+        self.rearArrow = self.createIndex(self.rear, "Rear", high=False)
+        self.frontArrow = self.createIndex(self.front, "Front", high=True)
```

Comment:
Let's use lowercase names for the labels to match the variable names in the textbook code, "front" and "rear".